### DISCUSSION TITLE
includeRedundantEnvironments missing from "new" fingerprint generators(?)

### DISCUSSION THREAD

**Discussion by jepdavidson** [rdkit/rdkit#5665] [Q&A]
Hi,

I think the question in the title more-or-less says it all; but I am not very familiar with the (relatively) new fingerprint generators, so I just wanted to double-check...  I'm using `2022.03.5  py310h1c297d8_0  conda-forge`.

In the olden days, if I wanted to generate some Morgan fingerprint keys and ignore redundancy I would do something like this:

```python
from rdkit import Chem
from rdkit.Chem import AllChem

mol = Chem.MolFromSmiles("C1CC1")
bi = {}
fp = AllChem.GetMorganFingerprint(mol, radius=2, includeRedundantEnvironments=True, bitInfo=bi)
bi
```
> {171200514: ((0, 2), (1, 2), (2, 2)),
> 2142032900: ((1, 1), (0, 1), (2, 1)),
> 2968968094: ((0, 0), (1, 0), (2, 0))}

So, now I'm moving with the times(!), I thought I would check out the 'new' code:

```python
from rdkit.Chem import rdFingerprintGenerator

fpgen = rdFingerprintGenerator.GetMorganGenerator(radius=2)
ao = rdFingerprintGenerator.AdditionalOutput()
ao.AllocateBitInfoMap()
# In python, no support for passing includeRedundantEnvironments argument
fp = fpgen.GetSparseCountFingerprint(Chem.MolFromSmiles("C1CCC1"), additionalOutput=ao)
ao.GetBitInfoMap()
```
> {171200514: ((0, 2),),
> 2142032900: ((1, 1), (0, 1), (2, 1)),
> 2968968094: ((0, 0), (1, 0), (2, 0))}

Am I missing a way of applying the `includeRedundantEnvironments` option, or is it currently not supported in via python?

Kind regards

James

**Comment by greglandrum** [comment]
Hi @jepdavidson : there are a number of “advanced “ options which haven’t made their way across to the new FP generators. This is likely one of them.

it would be great if you could file an enhancement request for this so that we have a reminder that it needs to be done
